cpp/include/cuspatial/traits.hpp

```diff
@@ -84,6 +87,18 @@ constexpr bool is_same_floating_point()
          std::conjunction_v<std::is_floating_point<Ts>...>;
 }
 
+template <typename T, typename Enable = void>
```

Comment:
Do you need to remove cvref qualifiers? https://stackoverflow.com/a/62757885/749748

Reply:
For completeness I'll add it. I also like this implementation better.

Reply:
The proposed still doesn't guard against `std::reference_wrapper<std::optional<int>>` though.